--- plasma-framework-5.24.0/src/declarativeimports/core/windowthumbnail.cpp	2016-09-13 21:02:09.536369625 +0600
+++ plasma-framework-5.24.0/src/declarativeimports/core/windowthumbnail.cpp	2016-09-13 21:29:12.260354213 +0600
@@ -276,7 +276,8 @@
                 return false;
             }
 
-            textureNode->reset(window()->createTextureFromId(m_texture, QSize(geo->width, geo->height), QQuickWindow::TextureOwnsGLTexture));
+            textureNode->reset(window()->createTextureFromId(m_texture, QSize(geo->width, geo->height)));
+            
         }
         textureNode->texture()->bind();
         bindGLXTexture();
@@ -325,7 +326,7 @@
                 size.setWidth(geo->width);
                 size.setHeight(geo->height);
             }
-            textureNode->reset(window()->createTextureFromId(m_texture, size, QQuickWindow::TextureOwnsGLTexture));
+            textureNode->reset(window()->createTextureFromId(m_texture, size));
         }
         textureNode->texture()->bind();
         bindEGLTexture();
@@ -647,14 +647,14 @@
         ((glXReleaseTexImageEXT_func)(m_releaseTexImage))(d, m_glxPixmap, GLX_FRONT_LEFT_EXT);
         glXDestroyPixmap(d, m_glxPixmap);
         m_glxPixmap = XCB_PIXMAP_NONE;
-        glDeleteTextures(1, &m_texture);
+        
     }
 #endif
 #if HAVE_EGL
     if (m_image != EGL_NO_IMAGE_KHR) {
         ((eglDestroyImageKHR_func)(m_eglDestroyImageKHR))(eglGetCurrentDisplay(), m_image);
         m_image = EGL_NO_IMAGE_KHR;
-        glDeleteTextures(1, &m_texture);
+        
     }
 #endif
     if (m_pixmap != XCB_WINDOW_NONE) {
