--- setup/Linux/oss/build/pci_wrapper.inc	2009-08-27 05:19:07.000000000 +0300
+++ setup/Linux/oss/build/pci_wrapper.inc	2013-02-08 13:56:10.519794975 +0200
@@ -17,7 +17,11 @@
 static dev_map_t dev_map[MAX_INSTANCE];
 static int n_devmap = 0;
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,8,0)
 static int __devinit
+#else
+static int
+#endif
 osspci_probe (struct pci_dev *pcidev, const struct pci_device_id *pciid)
 {
   oss_device_t *osdev;
@@ -65,7 +69,11 @@
   return 0;
 }
 
-static void __devexit
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,8,0)
+static int __devexit
+#else
+static int
+#endif
 osspci_remove (struct pci_dev *pcidev)
 {
   int i;
@@ -80,10 +88,11 @@
 	pci_disable_device (dev_map[i].pcidev);
 	osdev_delete (osdev);
 
-	return;
+	return 0;
       }
 
   printk (KERN_ALERT DRIVER_NICK ": Can't find the PCI device to detach\n");
+  return -EIO;
 }
 
 void

--- osscore.c.bak	2013-10-25 22:55:26.117237463 +0300
+++ /tmp/osscore.c	2013-10-27 20:08:27.773527712 +0200
@@ -280,6 +280,25 @@
   .read = oss_read_devfiles,
 };
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,10,0)
+static void
+init_proc_fs (void)
+{
+  if ((oss_proc_root =
+       proc_mkdir_mode ("opensound", 0700 | S_IFDIR, NULL)) == NULL)
+    {
+      oss_cmn_err (CE_CONT, "Cannot create /proc/opensound\n");
+      return;
+    }
+
+  if ((oss_proc_devfiles =
+       proc_create ("devfiles", 0600, oss_proc_root, &oss_proc_operations)) == NULL)
+    {
+      oss_cmn_err (CE_CONT, "Cannot create /proc/opensound/devfiles\n");
+      return;
+    }
+}
+#else
 static void
 init_proc_fs (void)
 {
@@ -299,6 +318,7 @@
 
   oss_proc_devfiles->proc_fops = &oss_proc_operations;
 }
+#endif
 
 static void
 uninit_proc_fs (void)
@@ -436,7 +456,9 @@
 int
 oss_get_uid (void)
 {
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,29)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,10,0)
+  return from_kuid_munged(current_user_ns(), current->cred->uid);
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,29)
   return current->cred->uid;
 #else
   return current->uid;
@@ -953,7 +975,9 @@
   fop->llseek = oss_no_llseek;
   fop->read = tmp_read;
   fop->write = tmp_write;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,11,0)
   fop->readdir = NULL;		/* tmp_readdir; */
+#endif
   fop->poll = tmp_poll;
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,35)
   fop->ioctl = tmp_ioctl;