--- a/bin/libqtile/widget/__init__.py	2018-08-20 09:49:29.935412605 +0600
+++ b/bin/libqtile/widget/__init__.py	2018-08-20 09:48:37.266411915 +0600
@@ -61,6 +61,9 @@
 safe_import(".notify", "Notify")
 safe_import(".prompt", "Prompt")
 safe_import(".sensors", "ThermalSensor")
+safe_import(".rgpusensors", "RGPUThermalSensor")
+safe_import(".chsensors", "ChThermalSensor")
+safe_import(".hddsensors", "HDThermalSensor")
 safe_import(".sep", "Sep")
 safe_import(".she", "She")
 safe_import(".spacer", "Spacer")
