--- src/core/mount-setup.c	2014-10-23 21:20:23.344118433 +0600
+++ src/core/mount-setup.c	2014-12-01 15:57:29.138001541 +0600
@@ -412,15 +412,12 @@
          * udevd. */
         dev_setup(NULL);
 
-        /* Mark the root directory as shared in regards to mount
-         * propagation. The kernel defaults to "private", but we think
-         * it makes more sense to have a default of "shared" so that
-         * nspawn and the container tools work out of the box. If
-         * specific setups need other settings they can reset the
-         * propagation mode to private if needed. */
+        /* The root directory marked as shared in regards to mount
+         * propagation. The kernel defaults to "private", but it is 
+	  * more safe revert it to private. */
         if (detect_container(NULL) <= 0)
-                if (mount(NULL, "/", NULL, MS_REC|MS_SHARED, NULL) < 0)
-                        log_warning("Failed to set up the root directory for shared mount propagation: %m");
+                if (mount(NULL, "/", NULL, MS_REC|MS_PRIVATE, NULL) < 0)
+                        log_warning("Failed to set up the root directory for private mount propagation: %m");
 
         /* Create a few directories we always want around, Note that
          * sd_booted() checks for /run/systemd/system, so this mkdir
